cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/validator/AwsIDBrokerObjectStorageValidator.java

```diff
@@ -0,0 +1,83 @@
+import com.sequenceiq.cloudbreak.cloud.model.filesystem.CloudS3View;
+import com.sequenceiq.cloudbreak.validation.ValidationResult;
+import com.sequenceiq.cloudbreak.validation.ValidationResult.ValidationResultBuilder;
+import com.sequenceiq.common.api.cloudstorage.AccountMappingBase;
+import com.sequenceiq.common.model.CloudIdentityType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+public class AwsIDBrokerObjectStorageValidator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsIDBrokerObjectStorageValidator.class);
+
+    private final AmazonIdentityManagement iam;
+
+    public AwsIDBrokerObjectStorageValidator(AmazonIdentityManagement amazonIdentityManagement) {
+        this.iam = amazonIdentityManagement;
+    }
+
+    public ValidationResult validateObjectStorage(SpiFileSystem spiFileSystem, ValidationResultBuilder resultBuilder) {
+        List<CloudFileSystemView> cloudFileSystems = spiFileSystem.getCloudFileSystems();
+        for (CloudFileSystemView cloudFileSystemView : cloudFileSystems) {
+            CloudS3View cloudFileSystem = (CloudS3View) cloudFileSystemView;
+            String instanceProfileArn = cloudFileSystem.getInstanceProfile();
+            InstanceProfile instanceProfile = AwsIamUtil.getInstanceProfile(iam, instanceProfileArn,
+                resultBuilder);
+            if (instanceProfile != null) {
+                CloudIdentityType cloudIdentityType = cloudFileSystem.getCloudIdentityType();
+                if (CloudIdentityType.ID_BROKER.equals(cloudIdentityType)) {
+                    validateIDBroker(instanceProfile, spiFileSystem, cloudFileSystem, resultBuilder);
+                } else if (CloudIdentityType.LOG.equals(cloudIdentityType)) {
+                    validateLog(instanceProfile, cloudFileSystem, resultBuilder);
+                }
+            }
+        }
+
+        return resultBuilder.build();
+    }
+
+    private void validateLog(InstanceProfile instanceProfile, CloudS3View cloudFileSystem,
+                                ValidationResultBuilder resultBuilder) {
+        AwsInstanceProfileEC2TrustValidator.isTrusted(instanceProfile, resultBuilder);
+
+        //TODO - need to figure out how to get LOGS_LOCATION_BASE value
+        //AwsLogRolePermissionValidator.validate(iam, instanceProfile, cloudFileSystem, resultBuilder);
+    }
+
+    private void validateIDBroker(InstanceProfile instanceProfile, SpiFileSystem spiFileSystem,
+                                    CloudS3View cloudFileSystem, ValidationResultBuilder resultBuilder) {
+        AwsInstanceProfileEC2TrustValidator.isTrusted(instanceProfile, resultBuilder);
+
+        Set<Role> allMappedRoles = getAllMappedRoles(cloudFileSystem, resultBuilder);
+        AwsIDBrokerAssumeRoleValidator.canAssumeRoles(iam, instanceProfile, allMappedRoles, resultBuilder);
```

Comment:
Some of the validators have static methods, some of them are instantiated. Neither of these are the preferred solution. I would suggest another approach:

In `AwsIDBrokerObjectStorageValidator` please make `AmazonIdentityManagement` a parameter of the `validateObjectStorage` method like here with the `AwsIDBrokerAssumeRoleValidator.canAssumeRoles` method. 
In all the validator classes make `iam, spiFileSystem` parameters of the `validate` method.

This way you won't have to instantiate any of these, because they will be stateless. And since we are writing stateless components, services, and not a collection on static methods, please make these methods non-static, and annotate them with @Component, and inject them wherever they are used.

Reply:
I think I addressed what you meant with the latest push.